aws/buckets/datalake.tf

```
@@ -0,0 +1,57 @@
+#
+# Holds exported data from ETL transformations
+#
+
+module "lake_bucket" {
+  source            = "github.com/cds-snc/terraform-modules//S3?ref=17994187b8628dc5decf74ead84768501378df4c" # ref for v10.0.0
+  bucket_name       = "cds-data-lake-bucket-${var.env}"
+  billing_tag_value = var.billing_tag_value
+
+  logging = {
+    target_bucket = module.log_bucket.s3_bucket_id
+    target_prefix = "datalake/"
+  }
+
+  lifecycle_rule = [
+    local.lifecycle_remove_noncurrent_versions,
+    local.lifecycle_transition_storage
+  ]
+
+  versioning = {
+    enabled = true
+  }
+}
+
+resource "aws_s3_bucket_policy" "lake_bucket" {
+  bucket = module.lake_bucket.s3_bucket_id
+  policy = data.aws_iam_policy_document.lake_bucket.json
+}
+
+#
+# There is a 20kb limit on the size of the policy document
+#
+data "aws_iam_policy_document" "lake_bucket" {
+  statement {
+    sid    = "CostAndUsageReport"
+    effect = "Allow"
+    principals {
+      type = "AWS"
+      identifiers = [
+        "arn:aws:iam::659087519042:role/BillingExtractTags",
+        "arn:aws:iam::659087519042:role/CostUsageReplicateToDataLake"
```

I think these will be replaced in the future with a Platform data lake IAM role that pulls in the data (unless we go with S3 replication, but that's future-us's problem).  For now though I think this bucket policy can also be removed. 
```suggestion
        "arn:aws:iam::659087519042:role/BillingExtractTags",
        "arn:aws:iam::659087519042:role/CostUsageReplicateToDataLake"
```